src/main.rs

```diff
@@ -122,16 +124,19 @@ fn use_interactive_shell() -> bool {
     if let Ok(interactive_flag) = env::var("WSLGIT_USE_INTERACTIVE_SHELL") {
         if interactive_flag == "false" || interactive_flag == "0" {
             return false;
+        } else {
+            return true;
         }
     }
-    // check for advanced usage indicated by BASH_ENV and WSLENV=BASH_ENV
+    // check for advanced usage indicated by BASH_ENV and WSLENV contains BASH_ENV
     else if env::var("BASH_ENV").is_ok() {
         if let Ok(wslenv) = env::var("WSLENV") {
-            if wslenv
-                .split(':')
-                .position(|r| r.eq_ignore_ascii_case("BASH_ENV"))
-                .is_some()
-            {
+            lazy_static! {
+                // BASH_ENV must be first or after a ':', and it must have flags
+                static ref BASH_ENV_RE: Regex = Regex::new(r"(?-u)(^|:)BASH_ENV/")
```

Comment:
Looks good, just a quick question. Is there any particular reason why you are requiring flags for the `BASH_ENV` value in `WSLENV`? As I understand it the flags are optional, and right now it only works without flags (due to limitations in wslgit - therefore I'm glad you improve this). To support all use cases, the regex could be something like
```
r"(?-u)(^|:)BASH_ENV($|/|:)"
```
to allow either nothing, flags or another item after `BASH_ENV`. [Here is a quick example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a0e447b6ccbaed379177d80d46487e9d) testing these configurations.

Reply:
You are right, the flags are not required. I'll fix it when I get home.